File: feraltweaks-server-module/src/main/java/org/asf/centuria/feraltweaks/api/versioning/IModVersionHandler.java
package org.asf.centuria.feraltweaks.api.versioning;

import java.util.Map;

/**
 * 
 * Interface for handling mod version handshake rules
 * 
 * @author Sky Swimmer
 *
 */
public interface IModVersionHandler {

	/**
	 * Retrieves the client mod handshake rules
	 * 
	 * @return Map of handshake rules expected for client mods (id and version check
	 *         string pairs)
	 */
	public Map<String, String> getClientModVersionRules();

}

File: feraltweaks/API/Versioning/IModVersionHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FeralTweaks.Versioning
{
    /// <summary>
    /// Interface for handling mod version handshake rules
    /// </summary>
    public interface IModVersionHandler
    {
        /// <summary>
        /// Retrieves the server mod handshake rules
        /// </summary>
        /// <returns>Map of handshake rules expected for server mods (id and version check string pairs)</returns>
        public Dictionary<string, string> GetServerModVersionRules();
    }
}
